tests/hypothesis_strategies.py

```diff
@@ -0,0 +1,158 @@
+"""Hypothesis strategies."""
+import random
+
+import hypothesis.strategies as st
+import numpy as np
+from hypothesis import assume
+from hypothesis.extra.pandas import columns, data_frames
+
+from baybe.exceptions import NumericalUnderflowError
+from baybe.parameters.categorical import (
+    CategoricalEncoding,
+    CategoricalParameter,
+    TaskParameter,
+)
+from baybe.parameters.custom import CustomDiscreteParameter
+from baybe.parameters.numerical import (
+    NumericalContinuousParameter,
+    NumericalDiscreteParameter,
+)
+from baybe.parameters.substance import SubstanceEncoding, SubstanceParameter
+from baybe.utils import DTypeFloatNumpy
+
+_largest_lower_interval = np.nextafter(
+    np.nextafter(np.inf, 0, dtype=DTypeFloatNumpy), 0, dtype=DTypeFloatNumpy
+)
+"""
+The largest possible value for the lower end of a continuous interval such that there
+still exists a larger but finite number for the upper interval end.
+"""
+
+_decorrelate = st.one_of(
+    st.booleans(),
+    st.floats(min_value=0.0, max_value=1.0, exclude_min=True, exclude_max=True),
+)
+"""A strategy that generates decorrelation settings."""
+
+parameter_name = st.text(min_size=1)
+"""A strategy that generates parameter names."""
+
+categories = st.lists(st.text(min_size=1), min_size=2, unique=True)
+"""A strategy that generates parameter categories."""
+
+
+@st.composite
+def smiles(draw: st.DrawFn):
+    """Generate short SMILES strings."""
```

Comment:
For the hypothesis part, I don't think this would be critical. Basically similar to my response to Alex when he asked about whether it's worth having more complex molecules: we only want to create something for our backend to digest that is close to the real use case.

However, for the backend part, this is worth a discussion. What if the user entered two smiles that are equivalent? Perhaps it's worth doing the additional check there by going via `canonical`?

So what I'm trying to say: for generation we don't necessarily need to bother, but for validation we perhaps should!? And if we do, THEN we would of course need to adjust the hypothesis strategy as well. 

Do you think this would cause some major delays in parameter creation? I guess our caching might save us here as well. Could you point me to that function, or even better, implement the check yourself, since you are the expert here?

Reply:
If the user enters two SMILES that describe the same molecule, this is a duplicate input, since what the user wants to use as a parameter is probably the *molecule described by the SMILES*, and not the SMILES. So in my opinion, this needs to be handled like every other part of the code in which we prevent the user from making multiple input (that is, unless there is any reason to have the same molecule twice with different representations *on purpose*, but I cannot think of this case).

So in my opinion, we need to either verify that a SMILES is canonical and provide a warning if we detect that the provided SMILES is not canonical or make everything canonical by ourselves. Since this is only a minor check (and should only be done once, right?!), I do not think that this creates an immense overhead.

Reply:
hmm im still undecided. We do not require that SMILES entered are canonical, is is generally not human readable whether a SMILES is canonical or not (so it would be quite a user experience downgrade) and we can work perfectly without canonical ones. However, IF we want to make sure all SMILES describe different molecules the way to go would be via creating the canonical smiles in the backend somewhere to perform the check. Overal thats a really niche issue that can be postponed to another PR. I was just worried that in the small space of molecules that you create here identical ones might be there causing problems down the line due to singularities (eg NCC and CCN)

Reply:
Update: I'm currently on it and will push a solution in the next minutes. Takeaways:
* The check will only happen in the converter, no need for the user to adjust anything
* Since our `is_valid_smiles` string already called rdkit anyway, there is no real additional load, I think
* I reworked the validator completely such that we now also get nice exceptiongroups (instead of stopping the program when encountering the first problematic input --> should to the same strategy in other validators when we touch then next time). Here a small taster:
  
<img width="635" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/7b02b52a-8daf-4dc2-b4fc-345f32446880">

<img width="821" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/d0f4d56e-988b-4216-8298-3dfa3fb9155d">

Reply:
Now online. If you like it, resolve. Otherwise, let me know :D

Reply:
Looks good to me, if @Scienfitz agrees we can resolve.